submissions/dashakim/oop-task/js/app.js

```diff
@@ -0,0 +1,125 @@
+var gameScore = 0;
+
+const ENEMY = {
+  speed: 100,
+  varianceSpeed: 222,
+  location: [70, 155, 235]
+};
+const PLAYER = {
+  startX: 200,
+  startY: 404,
+  stepX: 101,
+  stepY: 83
+};
+
+const CANVAS = {
+  begin: -60,
+  end: 510,
+  leftSide: 0,
+  rightSide: 400,
+  top: 0,
+  bottom: 404,
+  halfCell: 50
+};
+
+var Character = function(x, y, sprite) {
+  this.x = x;
+  this.y = y;
+  this.sprite = sprite;
+};
+Character.prototype.constructor = Character;
+Character.prototype.render = function() {
+  ctx.drawImage(Resources.get(this.sprite), this.x, this.y);
+};
+
+var Enemy = function(x, y, speed, player) {
+  Character.call(this, x, y);
+  this.sprite = "images/car.png";
+  this.speed = speed;
+  this.player = player;
+};
+Enemy.prototype = Object.create(Character.prototype);
+Enemy.prototype.constructor = Enemy;
+Enemy.prototype.toRender = function() {
+  this.render();
+};
+
+Enemy.prototype.update = function(dt) {
+  this.x += this.speed * dt;
+  if (this.x > CANVAS.end) {
+    this.x = CANVAS.begin;
+    //random speed for enemies
+    this.speed = ENEMY.speed + Math.floor(Math.random() * ENEMY.varianceSpeed);
+  }
+  // return to start position player when collide with enemy
+  if (
+    this.player.x > this.x - CANVAS.halfCell &&
+    this.player.x < this.x + CANVAS.halfCell &&
+    this.player.y > this.y - CANVAS.halfCell &&
+    this.player.y < this.y + CANVAS.halfCell
+  ) {
+    this.player.goToStart();
+  }
+};
+
+var Player = function(x, y, addScore) {
+  Character.call(this, x, y);
+  this.addScore = addScore;
+  this.sprite = "images/char-cat-girl.png";
```

Comment:
Please induce from the similar case with `Enemy` class.
Please do not expect peer reviewer to point at each line of code that requires fixes similar to those already done.

Reply:
I'm sorry, I didn't understand what I need to change. Oh, I feel so stupid)

Reply:
You mean to minus score after collision?

Reply:
I mean sprite. You passed it over to a parent class constructor in Enemy class and do not follow the same pattern with Player class